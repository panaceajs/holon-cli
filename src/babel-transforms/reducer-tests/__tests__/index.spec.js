const pluginTester = require('babel-plugin-tester');
const reducerTests = require('../');
const pluginSyntaxObjectRestSpread = require('@babel/plugin-syntax-object-rest-spread');
const dynamicImport = require('@babel/plugin-syntax-dynamic-import');

pluginTester({
  filename: __filename,
  plugin: reducerTests,
  snapshot: true,
  babelOptions: {
    sourceType: 'module',
    plugins: [dynamicImport, pluginSyntaxObjectRestSpread]
  },
  tests: [
    {
      title: 'it should change nothing when there is nothing to change.',
      fixture: '__fixtures__/without-describe-call.js',
      snapshot: false
    },
    {
      title: 'it should add tests for cases',
      fixture: '__fixtures__/with-describe-call.js',
      pluginOptions: { actionTypes: { doSomething: 'doSomething' } }
    },
    {
      title: 'it should update tests for cases',
      fixture: '__fixtures__/with-test.js',
      pluginOptions: {
        actionTypes: {
          doSomethingExisting: 'somethingExistingValue',
          doSomething: 'doSomething'
        }
      }
    }
  ]
});
