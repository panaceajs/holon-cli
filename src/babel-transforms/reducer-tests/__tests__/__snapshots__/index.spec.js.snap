// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`holon-reducer-tests it should add tests for cases: it should add tests for cases 1`] = `
"
describe('reducer', () => {});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('\`root\` reducer', () => {
  it('should handle \`DO_SOMETHING\`', () => {
    const next = reducer(undefined, doSomething());
    expect(next).toMatchSnapshot();
    const next2 = reducer(next, doSomething('do-something-unit-test-payload'));
    expect(next2).toMatchSnapshot();
  });
});
"
`;

exports[`holon-reducer-tests it should update tests for cases: it should update tests for cases 1`] = `
"
describe('reducer', () => {
  it('should handle \`DO_SOMETHING_EXISTING\`', () => {
    const next = reducer(undefined, doSomethingExisting());
    expect(next).toMatchSnapshot();
    const next2 = reducer(next, doSomething('do-something-existing-unit-test-payload'));
    expect(next2).toMatchSnapshot();
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('\`root\` reducer', () => {
  it('should handle \`DO_SOMETHING_EXISTING\`', () => {
    const next = reducer(undefined, doSomethingExisting());
    expect(next).toMatchSnapshot();
    const next2 = reducer(next, doSomething('do-something-existing-unit-test-payload'));
    expect(next2).toMatchSnapshot();
  });
  it('should handle \`DO_SOMETHING\`', () => {
    const next = reducer(undefined, doSomething());
    expect(next).toMatchSnapshot();
    const next2 = reducer(next, doSomething('do-something-unit-test-payload'));
    expect(next2).toMatchSnapshot();
  });
});
"
`;
